python/Ganga/GPIDev/Lib/File/LocalFile.py

```diff
@@ -224,6 +224,19 @@ def remove(self):
 
         return
 
+    def get(self, localPath=''):
```

Comment:
OK here is the problem in step by step detail.

I want to be able to run the following on all IGangaFile objects to get my file at some destination:
```
inputFile.localDir = tmpDir
inputFile.get()
```

I expect from an API that I get __ALL__ of the files in the tmpDir.

This will __NOT__ work with `LocalFile` due to the fact that `localDir` is used to point to the file in the first place.

Adding an exception for the `LocalFile` is against the whole point of this PR and means that we have some _funny_ behaviour with one of the objects which all share this same interface which just confuses me let alone future developers.

Using:
```
inputFile.get(localPath=tmpDir)
```

Allows the file regardless of the type to transfer a file here.
Allowing an optional parameter doesn't effect the default behaviour of the `get()` method being called and simply adds extra functionality.

A reasonable argument is that the file object should then have it's `localDir` replaced with the contents of `tmpdir`. In the case of `LocalFile` this represents a move command which is potentially dangerous so we shouldn't delete a user's file on them which raises the question of what is proper here?

Another possible solution is:

```
inputFile.transfer(localPath=tmpDir)
```

Which allows the `get()` functionality to be untouched but still leaves us with a question of which file is the LocalFile at this point?
Obviously different developers have valid opinions on this but ultimately the user pointed us to 1 file which we're copying for them.
We either lose information about 1 of the locations or expand the schema to allow this file to have multiple locations on disk which seems counter-intuitive.


Another solution is to yes change the `LocalFile` schema so that `localDir` is always only the place where the file will end up after `get()` and will be before `put()`.
We should really be using something like `location` for the construction of `LocalFile` in order to do this but this is an invasive possibly completely incompatible change.